InterfaceTPL/tplUnitTests.cpp

```diff
@@ -0,0 +1,139 @@
+ * IN WHOLE OR IN PART THE USE, STORAGE OR DISPOSAL OF THE SOFTWARE.
+ */
+
+#ifndef __TPL_UNIT_TESTS_CPP
+#define __TPL_UNIT_TESTS_CPP
+
+#include "tplUnitTests.hpp"
+#include "tsgBlasWrappers.hpp"
+#include <string>
+#include <math.h>
+#include <cmath>
+
+bool testTridiagonal();
+
+TplUnitTester::TplUnitTester() : verbose(false){}
+TplUnitTester::~TplUnitTester() {}
+
+void TplUnitTester::setVerbose(bool new_verbose){ verbose = new_verbose; }
+
+bool TplUnitTester::Test(UnitTests test){
+    if (test == unit_all) {
+        cout << endl << endl;
+        cout << "---------------------------------------------------------------------" << endl;
+        cout << "       Tasmanian Third Party Libraries (TPL): Unit Tests" << endl;
+        cout << "---------------------------------------------------------------------" << endl << endl;
+        bool pass = true;
+        pass = testTriToeplitzEigen();
+        cout << endl;
+        if (pass){
+            cout << "---------------------------------------------------------------------" << endl;
+            cout << "           All Unit Tests Completed Successfully" << endl;
+            cout << "---------------------------------------------------------------------" << endl << endl;
+        } else {
+            cout << "FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL" << endl;
+            cout << "         Some Unit Tests Have Failed" << endl;
+            cout << "FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL FAIL" << endl << endl;
+        }
+        return pass;
+    } else if (test == unit_none) {
+        cout << "No tests to run.\n";
+        return true;
+    } else {
+        cout << "Unknown test type!\n";
+        return false;
+    }
+}
+
+bool testTriToeplitzEigen() {
+    // Initialize.
+    const int N = 100;
+    bool is_matched;
+    double exact_eigs[N], D[N], E[N-1], W[N], Z[N*N], WORK1[2 * N - 2], WORK2[4 * N];
+    int IBLOCK1[N], ISPLIT1[N], IWORK1[3 * N];
+    double a = M_PI_2;
+    double b = M_PI_4;
+    ISPLIT1[0] = N;
+    IBLOCK1[0] = N;
+    for (int i=0; i<N; i++) {
+        exact_eigs[i] = a + 2.0 * b * cos((i+1) * M_PI / (N+1));
+    }
+    std::sort(exact_eigs, exact_eigs + N);
+
+    // Test LAPACK's dsterf function.
+    std::fill_n(D, N, a);
+    std::fill_n(E, N-1, b);
+    TasBLAS::sterf(N, D, E);
+    std::sort(D, D + N);
+    is_matched = doesMatch(N, D, exact_eigs);
+
+    // Test LAPACK's dsteqr function.
+    std::fill_n(D, N, a);
+    std::fill_n(E, N-1, b);
+    TasBLAS::steqr('N', N, D, E, Z, 1, WORK1);
+    std::sort(D, D + N);
+    is_matched = doesMatch(N, D, exact_eigs);
+
+    // Test LAPACK's dstebz function.
+    std::fill_n(D, N, a);
+    std::fill_n(E, N-1, b);
+    TasBLAS::stebz('A', 'E', N, 0.0, 0.0, 1, N, 1e-13, D, E, N, 1, W, IBLOCK1, ISPLIT1, WORK2, IWORK1);
+    is_matched = doesMatch(N, W, exact_eigs);
```

Comment:
What would happen if the call to `TasBLAS::steqr()` on line 100 fails, but the call to `TasBLAS::stebz()` on line 107 succeeds? 

If the QR method fails, then the unit test should fail, right?

Reply:
Yes. This logic has been fixed.